eth_account/_utils/structured_data/hashing.py

```diff
@@ -0,0 +1,319 @@
+                continue
+            elif field["type"] in deps:
+                # skip types that we have already encountered
+                continue
+            else:
+                # Custom Struct Type
+                struct_names_yet_to_be_expanded.append(field["type"])
+
+    # Don't need to make a struct as dependency of itself
+    deps.remove(primary_type)
+
+    return tuple(deps)
+
+
+def field_identifier(field):
+    """
+    Given a ``field`` of the format {'name': NAME, 'type': TYPE},
+    this function converts it to ``TYPE NAME``
+    """
+    return "{0} {1}".format(field["type"], field["name"])
+
+
+def encode_struct(struct_name, struct_field_types):
+    return "{0}({1})".format(
+        struct_name,
+        ','.join(map(field_identifier, struct_field_types)),
+    )
+
+
+def encode_type(primary_type, types):
+    """
+    The type of a struct is encoded as name ‖ "(" ‖ member₁ ‖ "," ‖ member₂ ‖ "," ‖ … ‖ memberₙ ")"
+    where each member is written as type ‖ " " ‖ name.
+    """
+    # Getting the dependencies and sorting them alphabetically as per EIP712
+    deps = get_dependencies(primary_type, types)
+    sorted_deps = (primary_type,) + tuple(sorted(deps))
+
+    result = ''.join(
+        [
+            encode_struct(struct_name, types[struct_name])
+            for struct_name in sorted_deps
+        ]
+    )
+    return result
+
+
+def hash_struct_type(primary_type, types):
+    return keccak(text=encode_type(primary_type, types))
+
+
+def is_valid_abi_type(type_name):
+    """
+    This function is used to make sure that the ``type_name`` is a valid ABI Type.
+
+    Please note that this is a temporary function and should be replaced by the corresponding
+    ABI function, once the following issue has been resolved.
+    https://github.com/ethereum/eth-abi/issues/125
+    """
+    valid_abi_types = {"address", "bool", "bytes", "int", "string", "uint"}
+    is_bytesN = type_name.startswith("bytes") and 1 <= int(type_name[5:]) <= 32
+    is_intN = (
+        type_name.startswith("int") and
+        8 <= int(type_name[3:]) <= 256 and
+        int(type_name[3:]) % 8 == 0
+    )
+    is_uintN = (
+        type_name.startswith("uint") and
+        8 <= int(type_name[4:]) <= 256 and
+        int(type_name[4:]) % 8 == 0
+    )
+
+    if type_name in valid_abi_types:
+        return True
+    elif is_bytesN:
+        # bytes1 to bytes32
+        return True
+    elif is_intN:
+        # int8 to int256
+        return True
+    elif is_uintN:
+        # uint8 to uint256
+        return True
+
+    return False
+
+
+def is_array_type(type):
+    # Identify if type such as "person[]" or "person[2]" is an array
+    abi_type = parse(type)
+    return abi_type.is_array
+
+
+@to_tuple
+def get_depths_and_dimensions(data, depth):
+    """
+    Yields 2-length tuples of depth and dimension of each element at that depth
+    """
+    if not isinstance(data, list) and not isinstance(data, tuple):
```

Comment:
This should be collapsed to a single check using `isinstance(data, (list, tuple))`.  If you want you can also use `eth_utils.is_list_like` though it is a bit more forgiving, allowing *any* `collections.abc.Sequence` type that is not a string type which probably isn't what we want here.